controllers/drplacementcontrol_controller.go

```diff
@@ -611,13 +611,18 @@ func (r *DRPlacementControlReconciler) Reconcile(ctx context.Context, req ctrl.R
 
 	ensureDRPCConditionsInited(&drpc.Status.Conditions, drpc.Generation, "Initialization")
 
-	placementObj, err := r.ownPlacementOrPlacementRule(ctx, drpc, logger)
+	placementObj, err := getPlacementOrPlacementRule(ctx, r.Client, drpc, logger)
 	if err != nil && !(errors.IsNotFound(err) && !drpc.GetDeletionTimestamp().IsZero()) {
 		r.recordFailure(drpc, placementObj, "Error", err.Error(), nil, logger)
 
 		return ctrl.Result{}, err
 	}
 
+	err = r.setDRPCOwnerAndAnnotatePlacement(ctx, drpc, placementObj, logger)
```

Comment:
`placementObj` can be nil here, if we got a NotFound error above. So a further nil check within `setDRPCOwnerAndAnnotatePlacement` maybe required?

Also we come here if PlacementObj is NotFound and DRPC is deleted, so we cannot add the owner ref in drpc anymore as it is deleted?

Reply:
Hmmm, There might be a small window where after we remove the finalizer from placement and update the placement successfully, we fail to remove the DRPC finalizer and update, or it just takes longer for it to persist. 
Ok, I'll move that line right after `isBeingDeleted` block without needing to check whether the placement is nil.